ui/app/mixins/with-route-visibility-detection.js

```diff
@@ -0,0 +1,12 @@
+import Mixin from '@ember/object/mixin';
+
+export default Mixin.create({
+  setupDocumentVisibility: function() {
+    this.set('_visibilityHandler', this.get('visibilityHandler').bind(this));
```

Comment:
I guess I have the same question about nested routes as I do with multiple components - do outer routes have their handler overwritten by child routes here?

Reply:
Nope, independent handlers again.